### PR TITLE
Some typos and nit fixes to the sections 1 and 2

### PR DESCRIPTION
**Pull request by CPerezz**

### CODE REVIEW

**Comment by CPerezz** [comment]
Also leaving here some concerns found while checking this:

- TxLog mentioned in evm-circuit.md is more advanced. Should we add a new section for it and add references to that?
- We talk about issue2.2 in section opcode-fetching but there’s no link to any place or explanation of what it is indeed.

File: src/architecture.md [hunk 1]
```diff
@@ -6,35 +6,31 @@
...
 
-However, sometimes there are some properties we can't ensure by only lookups. For example, the amount of all `item` should equal to the size of `table`, which is required by between EVM circuit and State circuit to prevent extra malicious write. In such case (`assert set(items) == table`), we need some extra constraint to ensure the relationship is correct. A naive approach is to also count all `item` in State circuit, which in the end is the size of the `table`, and constraint it to be equal to the one counted in EVM circuit.
+However, sometimes there are some properties we can't ensure only with lookups. For example, the amount of all `item` should equal to the size of `table`, which is required by between EVM circuit and State circuit to prevent extra malicious write. In such case (`assert set(items) == table`), we need some extra constraint to ensure the relationship is correct. A naive approach is to also count all `item` in State circuit, which in the end is the size of the `table`, and constraint it to be equal to the one counted in EVM circuit.
```
Comment (ed255) [inline src/architecture.md:33]: NOTE(Edu): Add a line explaining that halo2 lookup only proves that elements in lookup are subset of the table, and to prove that the table doesn't contain extra rows that are not looked-up we need to prove that the number of items in the table is as expected by counting (and matching the count value with the expected value)

File: src/architecture/evm-circuit/multi-step.md [hunk 2]
```diff
@@ -191,7 +191,7 @@ Take `CALL` then `CALLDATALOAD` as an example:
     - `dst_offset = memOffset = 0`
     - `bytes_left = length = 32`
 
-Then every step we check if `src_offset < src_end`, if not, we need to disalbe the source lookup and fill zeros into destination. Then add the `*_offset` by the amount of bytes we process at a step, and subtract `bytes_left` also by it, then propagate them to next step.
+Then, in every step we check if `src_offset < src_end`, if not, we need to disabe the source lookup and fill zeros into destination. Then add the `*_offset` by the amount of bytes we process at a step, and subtract `bytes_left` also by it, then propagate them to next step.
```
Comment (ed255) [inline src/architecture/evm-circuit/multi-step.md:194]:
```suggestion
Then, in every step we check if `src_offset < src_end`, if not, we need to disable the source lookup and fill zeros into destination. Then add the `*_offset` by the amount of bytes we process at a step, and subtract `bytes_left` also by it, then propagate them to next step.
```

File: src/architecture/evm-circuit/opcode-fetching.md [hunk 3]
```diff
@@ -108,9 +108,9 @@ Although it costs many steps to copy the creation code, it makes the opcode fetc
...
+When it comes to "memory copy", it means in EVM circuit we lookup both `rw_table` and `bytecode_table` to make sure the chunk of memory indeed exists in the latter table. However, EVM circuit doesn't have a friendly layout to do such operation (it costs many expressions to achieve so).
 
-If we want to further optimize "memory copy" respect to the concern in Step #2, since we know the memory to be copied is in chunk, and in `bytecode_table` it also exists in chunk, then we seem to let Bytecode circuit to do such operation with correct `rw_counter`, and in EVM circuit we only need to "trigger" such operation. We can add extra selector columns to enable it like:
+If we want to further optimize "memory copy" in respect to the concern hilighted in [Step #2](## Step #2 - Explicitly copy memory to bytecode_table), since we know the memory to be copied is in chunk, and in `bytecode_table` it also exists in chunk, then we seem to let Bytecode circuit to do such operation with correct `rw_counter`, and in EVM circuit we only need to "trigger" such operation. We can add extra selector columns to enable it like:
```
Comment (ed255) [inline src/architecture/evm-circuit/opcode-fetching.md:113]: The link should be `[Step #2](#step-2---explicitly-copy-memory-to-bytecode_table)`

**Review by ed255 — APPROVED** [review]
Thanks for the review and all the fixes!  I'll merge this and address myself the small details I found in a future commit.

**Pull request merged**